Add -a/--all option to doit forget

tbekolay opened the pull request:
I have a very large data analysis project that uses doit to analyze large data sets and cache the results of those analyses in pickle files, which are then loaded and used for plotting downstream. Doit handles this task perfectly. It's hard to overstate how much time doit has saved me when working on this project. 

However, one instance where doit has cost me time is when I have accidentally done `doit forget` with no arguments. I know that this forgets everything, but it is common for me to change a plotting task then forget it, and run it. When I'm tired, sometimes I press enter early when trying to forget the plotting task, which ends up forgetting tasks that cache analyses that take hours.

My solution to that is to force users to opt in to forgetting all tasks with an `-a/--all` flag (e.g., `doit forget --all`). I've implemented this in this PR, but I haven't yet filled out the changelog or added tests because I wanted to be sure that you would be okay with this change first @schettino72. If you're okay with it, then I will add those to this PR.

schettino72 commented:
Thanks Trevor. I understand your use-case. 

I guess in order to retain backward compatibility (and the functionality for those who like it) we would need to add one more command-line option `--disable-default-all`. For your use-case you would set this to True in your config and get the behavior you want. Makes sense?

tbekolay commented:
> we would need to add one more command-line option `--disable-default-all`

Yep that makes total sense. I'll make that change and add in some tests as soon as I can.

schettino72 commented:
Hey, I am planning a release this month. It would be great if you have time to complete this.

tbekolay commented:
Thanks @schettino72 I should be able to do this :+1:

tbekolay commented:
Okay I've updated this PR to add the `--disable-default-all` option, and updated metadata files where I could find them. I also added a note to the `forget` command documentation, but if you'd prefer to keep that short and let people discover the options through the `--help` system, feel free to remove the documentation note.

schettino72 commented:
This actually does not work the way I expected.

If you do not provide any parameter the `default_tasks` are forgotten, *not* all tasks...

```
DOIT_CONFIG = {
    'default_tasks': ['pyflakes', 'ut'],
}
```

So I will make following changes:

* rename `--disable-default-all` to `--disable-default`.
* `--all` will really forget *all* not just the default
* I will add an inverse `--enable-default` if your config has `--disable-default` but you want to explicitly forget the default tasks

Makes sense?

tbekolay commented:
Yes that works for me! In my use case I ran all tasks by default so I mistakenly thought that the default to was to forget all tasks rather than just the default tasks. Your suggestion makes sense. Let me know you want me to help with anything!

tbekolay commented:
Your redone version looks great @schettino72, thanks for getting this in the new release!
